app/controllers/CartController.php

```diff
@@ -0,0 +1,179 @@
+<?php
+namespace app\controllers;
+
+use app\exceptions\NoPermissionException;
+use DateTime;
+use PDO;
+
+class CartController
+{
+    protected PDO $db;
+    public function __construct(PDO $db)
+    {
+        $this->db = $db;
+    }
+
+    public function render($response)
+    {
+        return $response->setBody('app/views/CartView.php', [
+            'totalPrice' => $this->getTotalPrice()
+        ]);
+    }
+
+    public function add()
+    {
+        if(isset($_GET['id']))
+        {
+            $product = $this->db->prepare("SELECT * FROM products WHERE id=:id");
+            $product->execute([
+                'id'=> $_GET['id']
+            ]);
+            $product = $product->fetchAll(PDO::FETCH_CLASS);
+            if(!isset($_SESSION['cart']['products'][$_GET['id']]))
+            {
+                $_SESSION['cart']['products'][$_GET['id']] = $product;
+                $_SESSION['cart'][$_GET['id']]['quantity'] = 1;
+            }
+            else if($product[0]->quantity > $_SESSION['cart'][$_GET['id']]['quantity'])
+                $_SESSION['cart'][$_GET['id']]['quantity'] += 1;
+            header("Location: ../home");
+        }
```

Comment:
možda je ovako čitljivije

```suggestion
        $prodcutId = $_GET['id'] ?? null;
        if(!$prodcutId) {
            // nešto uraditi ako nema id
        }
        
        $product = $this->db->prepare("SELECT * FROM products WHERE id=:id");
            $product->execute([
                'id'=> $prodcutId
            ]);
            $product = $product->fetchAll(PDO::FETCH_CLASS);
            if(!isset($_SESSION['cart']['products'][$prodcutId]))
            {
                $_SESSION['cart']['products'][$prodcutId] = $product;
                $_SESSION['cart'][$prodcutId]['quantity'] = 1;
            }
            else if($product[0]->quantity > $_SESSION['cart'][$prodcutId]['quantity'])
                $_SESSION['cart'][$prodcutId]['quantity'] += 1;
            header("Location: ../home");
```